docs/api.adoc

```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
 
 Detailed description of api docs here!
 
-Hosts
+Search
 -----
-
 [nested]
-* List all
+* Search for host or group.
+Specify type as  a group, host, or any to limit the search.
 +
 [source]
 ....
-  /api/host/list
+GET /api/search?type=any&name=web_hosts
```

Comment:
based on a search term of "web_hosts", the below groups and hosts don't actually make any sense.  Let's modify the example response to show that this is a regex that will match.